File: 2Star AdvLvl/PALINPAIN.cpp
#include <iostream>
using namespace std;

void palindromes() {
    int a, b;
    cin >> a >> b;
    
	if (a%2==1 and b%2==1||b==1||a==1)
	    cout << -1 << endl;
	else if (a%2==0 and b%2==0) {
	    string s1(a/2,'a');
	    string s2(b,'b');
	    string s3(a,'a');
	    string s4(b/2,'b');
	    
	    cout << s1+s2+s1 << endl;
	    cout << s4+s3+s4 << endl;
	}
	else if (a%2==0) {
	    string s5(a/2,'a');
	    string s6(b,'b');
	    string s7(b/2,'b');
	    
	    cout << s5+s6+s5 << endl;
	    cout << s7+s5+'b'+s5+s7 << endl;
	}
	else if (b%2==0) {
	    string s8(b/2,'b');
	    string s9(a,'a');
	    string s10(a/2,'a');
	    
	    cout << s8+s9+s8 << endl;
	    cout << s10+s8+'a'+s8+s10 << endl;
    }
}

int main() {
	// your code goes here
	int t;
	cin >> t;
	while (t--) {
	    palindromes();
	}
	return 0;
}

File: 2Star AdvLvl/PALINPAIN.py
def palindromes():
    a, b = map(int, input().split())

    if (a % 2 == 1 and b % 2 == 1) or b == 1 or a == 1:
        print(-1)
    elif a % 2 == 0 and b % 2 == 0:
        s1 = 'a' * (a // 2)
        s2 = 'b' * b
        s3 = 'a' * a
        s4 = 'b' * (b // 2)

        print(s1 + s2 + s1)
        print(s4 + s3 + s4)
    elif a % 2 == 0:
        s5 = 'a' * (a // 2)
        s6 = 'b' * b
        s7 = 'b' * (b // 2)

        print(s5 + s6 + s5)
        print(s7 + s5 + 'b' + s5 + s7)
    elif b % 2 == 0:
        s8 = 'b' * (b // 2)
        s9 = 'a' * a
        s10 = 'a' * (a // 2)

        print(s8 + s9 + s8)
        print(s10 + s8 + 'a' + s8 + s10)

t = int(input())
for _ in range(t):
    palindromes()
